consensus/src/block_preparer.rs

```diff
@@ -43,7 +44,9 @@ impl BlockPreparer {
         tokio::task::spawn_blocking(move || {
             let filtered_txns = txn_filter.filter(block_id, block_timestamp_usecs, txns);
             let deduped_txns = txn_deduper.dedup(filtered_txns);
-            Ok(txn_shuffler.shuffle(deduped_txns))
+            let mut shuffled_txns = txn_shuffler.shuffle(deduped_txns);
+            shuffled_txns.truncate(max_txns_to_include_in_block);
```

Comment:
let's only do truncation if max_txns_to_include_in_block is not None

let's add a counter for to_execute_block_size from the result here

Reply:
Done